packages/desktop-client/src/components/reports/ReportOptions.tsx

```diff
@@ -44,3 +51,132 @@ const intervalOptions = [
 { value: 5, description: 'Yearly', name: 5,
 ];
 */
+export type QueryDataEntity = {
+  date: string;
+  category: string;
+  categoryGroup: string;
+  account: string;
+  accountOffBudget: boolean;
+  payee: string;
+  transferAccount: string;
+  amount: number;
+};
+
+export type UncategorizedEntity = {
+  id?: string;
+  name: string;
+  hidden?: boolean;
+  uncat_id: string;
+  offBudget: boolean;
+  transfer: boolean;
+  category: boolean;
+};
+
+export type UncategorizedGroupEntity = {
+  id?: string;
+  name: string;
+  hidden?: boolean;
+  categories?: UncategorizedEntity[];
+};
+
+let uncategorizedCategory: UncategorizedEntity = {
+  name: 'Uncategorized',
+  id: null,
+  uncat_id: '1',
+  hidden: false,
+  offBudget: false,
+  transfer: false,
+  category: false,
+};
+let transferCategory: UncategorizedEntity = {
+  name: 'Transfers',
+  id: null,
+  uncat_id: '2',
+  hidden: false,
+  transfer: true,
+  offBudget: false,
+  category: false,
+};
+let offBudgetCategory: UncategorizedEntity = {
+  name: 'Off Budget',
+  id: null,
+  uncat_id: '3',
+  hidden: false,
+  offBudget: true,
+  transfer: false,
+  category: true,
+};
+
+let uncategouncatGrouprizedGroup: UncategorizedGroupEntity = {
+  name: 'Uncategorized & Off Budget',
+  id: null,
+  hidden: false,
+  categories: [uncategorizedCategory, transferCategory, offBudgetCategory],
+};
+
+type categoryListsProps = {
+  hidden: boolean;
+  uncat: boolean;
+  categories: { list: CategoryEntity[]; grouped: CategoryGroupEntity[] };
+};
+export const categoryLists = ({
+  hidden,
+  uncat,
```

Comment:
<!-- Questions are appropriate if you have a potential concern but are not quite sure if it's relevant or not. Asking the author for clarification or investigation can lead to a quick resolution.  -->
**❓ question:** ‏what's `uncat`? Variable naming could be improved

Reply:
Would you be open to addressing the feedback in this PR instead? I really don't want to be jumping back-and-forth between multiple PRs for a single review.

Reply:
Renaming all these categories will nearly double the size of this PR. As previous discussed, this PR is already fairly large. I don't see the need to redo work I've already done. These variable names are not being introduced in this PR, they were already here from the original PR.

You don't need to jump back and forth between the PRs. Just don't worry about variable names for this PR as they will all be changed in an upcoming PR. My apologies for not making them better in the first place but I think the PR will become unmangeable if we try to add everything to it all at once.

Reply:
Pardon my ignorance, but I don't see `categoryLists` on `master` branch, so it is indeed new code. It's not existing functionality that's just moving to a different location.

As such: it gets the full scrutiny of a proper code review.

And just to clarify: I don't expect you to change `uncat` variable in the entire codebase. Just this new util function that you are introducing.

Reply:
No no, that's on me. I'd assumed you meant changing them for the entire code base, sorry for that. I've updated them and used the new variable names to match the 1996 PR for these variables.